mh/quest_utils.py

```diff
@@ -1548,921 +1745,63 @@ def read_quest_sm_data(fname):
-                (ItemsType.deviljho_gem, 1, 4),
-                (ItemsType.deviljho_scalp, 1, 8),
-                (ItemsType.deviljho_hide, 1, 17),
-                (ItemsType.hvy_armor_sphere, 1, 11),
-                (ItemsType.shining_charm, 1, 19),
-                (ItemsType.deviljho_fang, 1, 8)
-            ],
-            'rewards_row_2': [
-                (ItemsType.deviljho_scalp, 1, 20),
-                (ItemsType.timeworn_charm, 1, 34),
-                (ItemsType.deviljho_fang, 1, 6),
-                (ItemsType.deviljho_hide, 1, 17),
-                (ItemsType.deviljho_gem, 1, 3),
-                (ItemsType.shining_charm, 1, 11),
-                (ItemsType.hvy_armor_sphere, 1, 9)
-            ],
-        },
-        'subquest_1': {
-            'description': "None",
-            'type': 0x00000000,
-            'objective_type': Monster.none,
-            'objective_num': 0x00,
-            'zenny_reward': 0,
-            'hrp_reward': 0x00000000,
-            'rewards_row_1': [],
-        },
-        'subquest_2': {
-            'description': "None",
-            'type': 0x00000000,
-            'objective_type': Monster.none,
-            'objective_num': 0x00,
-            'zenny_reward': 0,
-            'hrp_reward': 0x00000000,
-            'rewards_row_1': [],
-        },
-    },
-    'unknown': {
-        # 2 for large mon quest, 3 for small/delivery, 5 for jhen/ala
-        'unk_12': 0x00000002,
-        'unk_4': 0x00,
-        'unk_5': 0x00,
-        'unk_6': 0x00,
-        'unk_7': 0x00000000,
-        'unk_9': 0x00000000,
-        'unk_10': 0x00000000,
-        'unk_11': 0x00000000,
-    }
-}
+def byteify(input):
+    if isinstance(input, dict):
+        return {byteify(key): byteify(value)
+                for key, value in input.iteritems()}
+    elif isinstance(input, list):
+        return [byteify(element) for element in input]
+    elif isinstance(input, unicode):
+        return input.encode('utf-8')
+    else:
+        return input
+
+
+class QuestLoader:
+    def __init__(self, file_loc='event/quest_rotation.json'):
+        self.version = 0
+        self.version_incrementer = 28
+        self.file_loc = file_loc
+        self.load_quests()
+        self.version = 0
+
+    def load_quests(self, file_loc=None):
+        import json
+        self.version += self.version_incrementer
+        quest_list = []
+        with open(self.file_loc, 'r') as f:
+            quests_json = byteify(json.load(f))
+            for day in quests_json:
+                curr_quest_list = []
+                for quest_file in day:
+                    with open(quest_file, 'r') as ff:
+                        curr_quest = byteify(json.load(ff))
+                        curr_quest = make_event_slot(curr_quest)
+                        curr_quest_list.append(curr_quest)
+                quest_list.append(curr_quest_list)
+        self.quests = quest_list
+    
+    def get_quest(self, idx):
+        return self.quests(idx)
```

Comment:
- Is this ever used?
- Isn't `self.quests` a list rather than a callable?

Reply:
This has been removed